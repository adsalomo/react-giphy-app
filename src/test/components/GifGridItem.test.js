import React from 'react'
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";
import '@testing-library/jest-dom';

describe('Prueba GifGridItem', () => {

    const url = 'http://www.adsalomo.com';
    const title = 'Hola mundo';

    const wrapper = shallow(
        <GifGridItem
            title={title}
            url={url}
        />
    );

    test('Debe de mostrar la foto del componente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe contener el titulo en el <p></p>', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('La imagen debe tener la URL y el titulo alt', () => {
        const img = wrapper.find('img');
        //console.log(img.props()); // propiedades de la img, componentes de html
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('Debe tener la clase de estilos: animate__fadeIn', () => {
        const div = wrapper.find('div');

        //console.log(div.prop('className'));

        expect(div.prop('className')).toContain('animate__fadeIn');
    });



});
