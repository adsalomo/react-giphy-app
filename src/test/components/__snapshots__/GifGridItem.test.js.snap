// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba GifGridItem Debe de mostrar la foto del componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Hola mundo"
    src="http://www.adsalomo.com"
  />
  <p>
     
    Hola mundo
     
  </p>
</div>
`;
